Trilinos nightly failures: cuda/11.4.2, incompatible change with Sacado package

**Describe the bug**
Nightly Trilinos builds with cuda/11.4.2+gcc/8.5.0 fail to compile after recent updates, breaking with Sacado:

```
20:44:36 /usr/include/c++/8/type_traits(2287): error: class "std::enable_if<false, Sacado::Fad::Exp::GeneralFad<Sacado::Fad::Exp::ViewStorage<double, 5U, 0U, Sacado::Fad::Exp::GeneralFad<Sacado::Fad::Exp::StaticFixedStorage<double, 5>>>>>" has no member "type"
20:44:36           detected during:
20:44:36             instantiation of type "std::enable_if_t<false, Sacado::Fad::Exp::ViewFad<double, 5U, 0U, Sacado::Fad::Exp::GeneralFad<Sacado::Fad::Exp::StaticFixedStorage<double, 5>>>>" 
20:44:36 /home/jenkins/weaver/workspace/KokkosEco_Trilinos_Weaver_CUDA112_Shared_Cplx/Trilinos/kokkos/containers/src/Kokkos_DynRankView.hpp(591): here
20:44:36             instantiation of class "Kokkos::DynRankView<DataType, Properties...> [with DataType=SFadType, Properties=<Kokkos::LayoutLeft, Kokkos::Serial>]" 
20:44:36 /home/jenkins/weaver/workspace/KokkosEco_Trilinos_Weaver_CUDA112_Shared_Cplx/Trilinos/packages/sacado/test/UnitTests/Fad_KokkosTests.hpp(1399): here
20:44:36             instantiation of "void Kokkos_View_Fad_DynRankDimensionScalar_UnitTest<FadType, Layout, Device>::runUnitTestImpl(Teuchos::FancyOStream &, __nv_bool &) const [with FadType=SFadType, Layout=Kokkos::LayoutLeft, Device=Kokkos::Serial]" 
20:44:36 /home/jenkins/weaver/workspace/KokkosEco_Trilinos_Weaver_CUDA112_Shared_Cplx/Trilinos/packages/sacado/test/UnitTests/Fad_KokkosTests_Serial.cpp(19): here
20:44:36 
20:44:36 /home/jenkins/weaver/workspace/KokkosEco_Trilinos_Weaver_CUDA112_Shared_Cplx/Trilinos/packages/sacado/test/UnitTests/Fad_KokkosTests.hpp(1483): error: more than one instance of overloaded function "Kokkos::DynRankView<DataType, Properties...>::operator() [with DataType=Sacado::Fad::SFad<double, 5>, Properties=<std::conditional_t<true, Kokkos::LayoutLeft, Kokkos::Serial::array_layout>, std::conditional_t<true, Kokkos::Serial::memory_space, std::conditional_t<true, Kokkos::Serial, std::conditional_t<true, Kokkos::Device<Kokkos::HostSpace::execution_space, Kokkos::HostSpace::memory_space>, Kokkos::HostSpace>>>>]" matches the argument list:
20:44:36             function template "std::enable_if_t<false, Sacado::Fad::Exp::ViewFad<double, 5U, 0U, Sacado::Fad::Exp::GeneralFad<Sacado::Fad::Exp::StaticFixedStorage<double, 5>>>> Kokkos::DynRankView<DataType, Properties...>::operator()(const iType &) const [with DataType=Sacado::Fad::SFad<double, 5>, Properties=<std::conditional_t<true, Kokkos::LayoutLeft, Kokkos::Serial::array_layout>, std::conditional_t<true, Kokkos::Serial::memory_space, std::conditional_t<true, Kokkos::Serial, std::conditional_t<true, Kokkos::Device<Kokkos::HostSpace::execution_space, Kokkos::HostSpace::memory_space>, Kokkos::HostSpace>>>>]"
20:44:36             function template "std::enable_if_t<true, Sacado::Fad::Exp::ViewFad<double, 5U, 0U, Sacado::Fad::Exp::GeneralFad<Sacado::Fad::Exp::StaticFixedStorage<double, 5>>>> Kokkos::DynRankView<DataType, Properties...>::operator()(const iType &) const [with DataType=Sacado::Fad::SFad<double, 5>, Properties=<std::conditional_t<true, Kokkos::LayoutLeft, Kokkos::Serial::array_layout>, std::conditional_t<true, Kokkos::Serial::memory_space, std::conditional_t<true, Kokkos::Serial, std::conditional_t<true, Kokkos::Device<Kokkos::HostSpace::execution_space, Kokkos::HostSpace::memory_space>, Kokkos::HostSpace>>>>]"
20:44:36             argument types are: (size_type)
20:44:36             object type is: Kokkos::DynRankView<Sacado::Fad::SFad<double, 5>, std::conditional_t<true, Kokkos::LayoutLeft, Kokkos::Serial::array_layout>, std::conditional_t<true, Kokkos::Serial::memory_space, std::conditional_t<true, Kokkos::Serial, std::conditional_t<true, Kokkos::Device<Kokkos::HostSpace::execution_space, Kokkos::HostSpace::memory_space>, Kokkos::HostSpace>>>>
20:44:36           detected during instantiation of "void Kokkos_View_Fad_DynRankAssignStatic1_UnitTest<FadType, Layout, Device>::runUnitTestImpl(Teuchos::FancyOStream &, __nv_bool &) const [with FadType=SFadType, Layout=Kokkos::LayoutLeft, Device=Kokkos::Serial]" 
20:44:36 /home/jenkins/weaver/workspace/KokkosEco_Trilinos_Weaver_CUDA112_Shared_Cplx/Trilinos/packages/sacado/test/UnitTests/Fad_KokkosTests_Serial.cpp(19): here
...
```

These are the changes that merged since the previous passing build:

**Changes**
    Fix misspelled cmake variable, for some systems leading to not compiling all tests — [Christian Robert Trott](https://jenkins-son.sandia.gov/user/crtrott/) / [detail](https://jenkins-son.sandia.gov/job/KokkosEco_Trilinos_Weaver_CUDA112_Shared_Cplx/707/changes#detail0)
    Fix overlooked naming test in #7222 — [Christian Robert Trott](https://jenkins-son.sandia.gov/user/crtrott/) / [detail](https://jenkins-son.sandia.gov/job/KokkosEco_Trilinos_Weaver_CUDA112_Shared_Cplx/707/changes#detail1)
    Fixup [Experimental::]SYCL — [dalg24](https://jenkins-son.sandia.gov/user/dalg24/) / [detail](https://jenkins-son.sandia.gov/job/KokkosEco_Trilinos_Weaver_CUDA112_Shared_Cplx/707/changes#detail2)
    Use macros to protect use of SharedSpace — [Christian Robert Trott](https://jenkins-son.sandia.gov/user/crtrott/) / [detail](https://jenkins-son.sandia.gov/job/KokkosEco_Trilinos_Weaver_CUDA112_Shared_Cplx/707/changes#detail3)
    core(graph): promote `instantiate` to public API — [romin.tomasetti](https://jenkins-son.sandia.gov/user/romin.tomasetti/) / [detail](https://jenkins-son.sandia.gov/job/KokkosEco_Trilinos_Weaver_CUDA112_Shared_Cplx/707/changes#detail4)
    Enforce modernize-type-traits (#7227) — [github](https://jenkins-son.sandia.gov/user/github/) / [detail](https://jenkins-son.sandia.gov/job/KokkosEco_Trilinos_Weaver_CUDA112_Shared_Cplx/707/changes#detail5)
    typo: kokkkos -> kokkos — [romin.tomasetti](https://jenkins-son.sandia.gov/user/romin.tomasetti/) / [detail](https://jenkins-son.sandia.gov/job/KokkosEco_Trilinos_Weaver_CUDA112_Shared_Cplx/707/changes#detail6)
    core(view): aligning `HostMirror` and `host_mirror_type` — [romin.tomasetti](https://jenkins-son.sandia.gov/user/romin.tomasetti/) / [detail](https://jenkins-son.sandia.gov/job/KokkosEco_Trilinos_Weaver_CUDA112_Shared_Cplx/707/changes#detail7)
    Add support for CUDA unified memory architectures i.e. Grace Hopper (#6823) — [github](https://jenkins-son.sandia.gov/user/github/) / [detail](https://jenkins-son.sandia.gov/job/KokkosEco_Trilinos_Weaver_CUDA112_Shared_Cplx/707/changes#detail8)
    Using CUDA limits to set extents for blocks,grids (#7235) — [github](https://jenkins-son.sandia.gov/user/github/) / [detail](https://jenkins-son.sandia.gov/job/KokkosEco_Trilinos_Weaver_CUDA112_Shared_Cplx/707/changes#detail9)
    Fix minor typing error. (#2314) — [github](https://jenkins-son.sandia.gov/user/github/) / [detail](https://jenkins-son.sandia.gov/job/KokkosEco_Trilinos_Weaver_CUDA112_Shared_Cplx/707/changes#detail10)

**Please include the following for a minimal reproducer**
**Reproducer** (weaver testbed, rhel8 queue):
```bash
# Repo prep
git clone -b develop https://github.com/trilinos/Trilinos.git
TRILINOS_DIR=$PWD/Trilinos
git clone -b develop https://github.com/kokkos/kokkos.git
KOKKOS_DIR=$PWD/kokkos
git clone -b develop https://github.com/kokkos/kokkos-kernels.git
KOKKOSKERNELS_DIR=$PWD/kokkos-kernels

cd $TRILINOS_DIR
ln -s ${KOKKOS_DIR}/kokkos kokkos
ln -s ${KOKKOSKERNELS_DIR}/kokkos-kernels kokkos-kernels
cd -

mkdir -p Build
cd Build

# Setup environment
bsub -Is -n 1 -q rhel8 -gpu "num=4" bash

source /etc/profile.d/modules.sh
source /projects/ppc64le-pwr9-rhel8/legacy-env.sh
module purge
module load git cmake/3.23.1 cuda/11.2.2/gcc/8.3.1 openmpi/4.1.1/gcc/8.3.1/cuda/11.2.2 openblas/0.3.18/gcc/8.3.1
module list
export OMPI_CXX=$KOKKOS_DIR/bin/nvcc_wrapper

# Configuration
cmake \
 -D CMAKE_INSTALL_PREFIX=${PWD}/install \
 -D CMAKE_CXX_COMPILER="`which mpicxx`" \
 -D CMAKE_C_COMPILER="`which mpicc`" \
 -D CMAKE_Fortran_COMPILER="`which mpifort`" \
 -D CMAKE_BUILD_TYPE:STRING=RELEASE \
 -D CMAKE_CXX_STANDARD=17 \
 -D BUILD_SHARED_LIBS=ON \
 -D CMAKE_INSTALL_PREFIX=$PWD/install \
 -D TPL_ENABLE_CUDA:STRING=ON \
 -D TPL_ENABLE_MPI:STRING=ON \
 -D MPI_BASE_DIR:PATH="$OPENMPI_ROOT" \
 -D MPI_BIN_DIR:PATH="$OPENMPI_BIN" \
 -D MPI_EXEC_POST_NUMPROCS_FLAGS:STRING="-map-by;socket:PE=4" \
-D TPL_ENABLE_BLAS:STRING=ON \
  -D BLAS_LIBRARY_DIRS:FILEPATH="$OPENBLAS_ROOT/lib" \
  -D BLAS_LIBRARY_NAMES:STRING="openblas" \
-D TPL_ENABLE_LAPACK:STRING=ON \
  -D LAPACK_INCLUDE_DIRS:FILEPATH="$OPENBLAS_ROOT/include" \
  -D LAPACK_LIBRARY_DIRS:FILEPATH="$OPENBLAS_ROOT/lib" \
  -D LAPACK_LIBRARY_NAMES:STRING="openblas" \
 -D Trilinos_ENABLE_COMPLEX=ON \
 -D Trilinos_ENABLE_TESTS=OFF \
 -D Trilinos_ENABLE_EXAMPLES=OFF \
 -D Trilinos_ENABLE_Kokkos=ON \
 -D Kokkos_ENABLE_CUDA=ON \
 -D Kokkos_ENABLE_CUDA_LAMBDA=ON \
 -D Kokkos_ENABLE_CUDA_UVM=ON \
 -D Kokkos_ARCH_VOLTA70=ON \
 -D Kokkos_ARCH_POWER9=ON \
 -D Kokkos_ENABLE_IMPL_CUDA_MALLOC_ASYNC=OFF \
 -D Kokkos_ENABLE_TESTS=ON \
 -D Kokkos_ENABLE_IMPL_MDSPAN=ON \
 -D Trilinos_ENABLE_KokkosKernels=ON \
 -D KokkosKernels_INST_MEMSPACE_CUDAUVMSPACE=ON \
 -D KokkosKernels_ENABLE_TPL_CUSOLVER=ON \
 -D KokkosKernels_ENABLE_TESTS=ON \
 -D Trilinos_ENABLE_Tpetra=ON \
 -D Tpetra_ALLOCATE_IN_SHARED_SPACE=ON \
 -D Tpetra_INST_CUDA=ON \
 -D Tpetra_INST_SERIAL=ON \
 -D Tpetra_ENABLE_TESTS=ON \
 -D Trilinos_ENABLE_Sacado=ON \
 -D Sacado_ENABLE_TESTS=ON \
 -D Trilinos_ENABLE_Phalanx=ON \
 -D Phalanx_ENABLE_TESTS=ON \
 -D Trilinos_ENABLE_Adelus=ON \
 -D Adelus_ENABLE_TESTS=ON \
 -D Trilinos_ENABLE_Ifpack2=ON \
 -D Ifpack2_ENABLE_TESTS=ON \
 -D Trilinos_ENABLE_Amesos2=OFF \
 -D Trilinos_ENABLE_Stokhos=OFF \
 -D Trilinos_ENABLE_MueLu=OFF \
 -D Intrepid2_ENABLE_TESTS=OFF \
 -D Trilinos_ENABLE_ShyLU_NodeTacho=OFF \
 -D Kokkos_CoreUnitTest_Default_MPI_1_EXTRA_ARGS="--gtest_filter=-*defaultdevicetype.shared_space" \
  -DTrilinos_ENABLE_INSTALLATION_TESTING=OFF \
  -DKokkos_SOURCE_DIR_OVERRIDE:STRING=kokkos \
  -DKokkosKernels_SOURCE_DIR_OVERRIDE:STRING=kokkos-kernels \
$TRILINOS_DIR

make -j16
```

* Note: only Sacado and tests likely need to be enabled to reproduce, complete job reproducer is posted

Build errors in nightly gcc/11.3.0 Serial build of Trilinos as well:

```
20:31:24 In file included from /home/jenkins/blake-new/workspace/KokkosEco_Trilinos_Blake_Gcc_1130_OpenMPI_415_Serial/Trilinos/packages/sacado/test/UnitTests/Fad_KokkosTests_Serial.cpp:15:
20:31:24 /home/jenkins/blake-new/workspace/KokkosEco_Trilinos_Blake_Gcc_1130_OpenMPI_415_Serial/Trilinos/packages/sacado/test/UnitTests/Fad_KokkosTests.hpp: In instantiation of 'void Kokkos_View_Fad_Resize_UnitTest<FadType, Layout, Device>::runUnitTestImpl(Teuchos::FancyOStream&, bool&) const [with FadType = Sacado::Fad::Exp::GeneralFad<Sacado::Fad::Exp::StaticFixedStorage<double, 5> >; Layout = Kokkos::LayoutContiguous<Kokkos::LayoutLeft>; Device = Kokkos::Serial; Teuchos::FancyOStream = Teuchos::basic_FancyOStream<char>]':
20:31:24 /home/jenkins/blake-new/workspace/KokkosEco_Trilinos_Blake_Gcc_1130_OpenMPI_415_Serial/Trilinos/packages/sacado/test/UnitTests/Fad_KokkosTests_Serial.cpp:19:1:   required from here
20:31:24 /home/jenkins/blake-new/workspace/KokkosEco_Trilinos_Blake_Gcc_1130_OpenMPI_415_Serial/Trilinos/packages/sacado/test/UnitTests/Fad_KokkosTests.hpp:958:17: error: no matching function for call to 'resize(ViewType&, const size_type&, long unsigned int, long unsigned int)'
20:31:24   958 |   Kokkos::resize(v, num_rows, num_cols+1, fad_size+1);
20:31:24       |   ~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
20:31:24 In file included from /home/jenkins/blake-new/workspace/KokkosEco_Trilinos_Blake_Gcc_1130_OpenMPI_415_Serial/Trilinos/kokkos/core/src/Kokkos_Core.hpp:68,
20:31:24                  from /home/jenkins/blake-new/workspace/KokkosEco_Trilinos_Blake_Gcc_1130_OpenMPI_415_Serial/Trilinos/packages/sacado/src/KokkosExp_View_Fad.hpp:112,
20:31:24                  from /home/jenkins/blake-new/workspace/KokkosEco_Trilinos_Blake_Gcc_1130_OpenMPI_415_Serial/Trilinos/packages/sacado/src/new_design/Sacado_Fad_Exp_GeneralFadTraits.hpp:197,
20:31:24                  from /home/jenkins/blake-new/workspace/KokkosEco_Trilinos_Blake_Gcc_1130_OpenMPI_415_Serial/Trilinos/packages/sacado/src/Sacado.hpp:32,
20:31:24                  from /home/jenkins/blake-new/workspace/KokkosEco_Trilinos_Blake_Gcc_1130_OpenMPI_415_Serial/Trilinos/packages/sacado/test/UnitTests/Fad_KokkosTests.hpp:12,
20:31:24                  from /home/jenkins/blake-new/workspace/KokkosEco_Trilinos_Blake_Gcc_1130_OpenMPI_415_Serial/Trilinos/packages/sacado/test/UnitTests/Fad_KokkosTests_Serial.cpp:15:
20:31:24 /home/jenkins/blake-new/workspace/KokkosEco_Trilinos_Blake_Gcc_1130_OpenMPI_415_Serial/Trilinos/kokkos/core/src/Kokkos_CopyViews.hpp:3043:1: note: candidate: 'template<class T, class ... P, class ... ViewCtorArgs> std::enable_if_t<(is_same_v<typename Kokkos::View<DataType, Properties>::array_layout, Kokkos::LayoutLeft> || is_same_v<typename Kokkos::View<DataType, Properties>::array_layout, Kokkos::LayoutRight>)> Kokkos::resize(const Kokkos::Impl::ViewCtorProp<ViewCtorArgs ...>&, Kokkos::View<D, P ...>&, size_t, size_t, size_t, size_t, size_t, size_t, size_t, size_t)'
20:31:24  3043 | resize(const Impl::ViewCtorProp<ViewCtorArgs...>& arg_prop,
20:31:24       | ^~~~~~
20:31:24 /home/jenkins/blake-new/workspace/KokkosEco_Trilinos_Blake_Gcc_1130_OpenMPI_415_Serial/Trilinos/kokkos/core/src/Kokkos_CopyViews.hpp:3043:1: note:   template argument deduction/substitution failed:
20:31:24 In file included from /home/jenkins/blake-new/workspace/KokkosEco_Trilinos_Blake_Gcc_1130_OpenMPI_415_Serial/Trilinos/packages/sacado/test/UnitTests/Fad_KokkosTests_Serial.cpp:15:
20:31:24 /home/jenkins/blake-new/workspace/KokkosEco_Trilinos_Blake_Gcc_1130_OpenMPI_415_Serial/Trilinos/packages/sacado/test/UnitTests/Fad_KokkosTests.hpp:958:17: note:   'ViewType' {aka 'Kokkos::View<Sacado::Fad::Exp::GeneralFad<Sacado::Fad::Exp::StaticFixedStorage<double, 5> >**, Kokkos::LayoutContiguous<Kokkos::LayoutLeft, 1>, Kokkos::Serial>'} is not derived from 'const Kokkos::Impl::ViewCtorProp<ViewCtorArgs ...>'
20:31:24   958 |   Kokkos::resize(v, num_rows, num_cols+1, fad_size+1);
20:31:24       |   ~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
20:31:24 In file included from /home/jenkins/blake-new/workspace/KokkosEco_Trilinos_Blake_Gcc_1130_OpenMPI_415_Serial/Trilinos/kokkos/core/src/Kokkos_Core.hpp:68,
20:31:24                  from /home/jenkins/blake-new/workspace/KokkosEco_Trilinos_Blake_Gcc_1130_OpenMPI_415_Serial/Trilinos/packages/sacado/src/KokkosExp_View_Fad.hpp:112,
20:31:24                  from /home/jenkins/blake-new/workspace/KokkosEco_Trilinos_Blake_Gcc_1130_OpenMPI_415_Serial/Trilinos/packages/sacado/src/new_design/Sacado_Fad_Exp_GeneralFadTraits.hpp:197,
20:31:24                  from /home/jenkins/blake-new/workspace/KokkosEco_Trilinos_Blake_Gcc_1130_OpenMPI_415_Serial/Trilinos/packages/sacado/src/Sacado.hpp:32,
20:31:24                  from /home/jenkins/blake-new/workspace/KokkosEco_Trilinos_Blake_Gcc_1130_OpenMPI_415_Serial/Trilinos/packages/sacado/test/UnitTests/Fad_KokkosTests.hpp:12,
20:31:24                  from /home/jenkins/blake-new/workspace/KokkosEco_Trilinos_Blake_Gcc_1130_OpenMPI_415_Serial/Trilinos/packages/sacado/test/UnitTests/Fad_KokkosTests_Serial.cpp:15:
20:31:24 /home/jenkins/blake-new/workspace/KokkosEco_Trilinos_Blake_Gcc_1130_OpenMPI_415_Serial/Trilinos/kokkos/core/src/Kokkos_CopyViews.hpp:3061:1: note: candidate: 'template<class T, class ... P> std::enable_if_t<(is_same_v<typename Kokkos::View<DataType, Properties>::array_layout, Kokkos::LayoutLeft> || is_same_v<typename Kokkos::View<DataType, Properties>::array_layout, Kokkos::LayoutRight>)> Kokkos::resize(Kokkos::View<D, P ...>&, size_t, size_t, size_t, size_t, size_t, size_t, size_t, size_t)'
20:31:24  3061 | resize(Kokkos::View<T, P...>& v, const size_t n0 = KOKKOS_IMPL_CTOR_DEFAULT_ARG,
20:31:24       | ^~~~~~
20:31:24 /home/jenkins/blake-new/workspace/KokkosEco_Trilinos_Blake_Gcc_1130_OpenMPI_415_Serial/Trilinos/kokkos/core/src/Kokkos_CopyViews.hpp:3061:1: note:   template argument deduction/substitution failed:
```

--- a/containers/src/Kokkos_DynRankView.hpp
+++ b/containers/src/Kokkos_DynRankView.hpp
@@ -584,12 +584,12 @@ class DynRankView : public ViewTraits<DataType, Properties...> {
   // This assumes a contiguous underlying memory (i.e. no padding, no
   // striding...)
   template <typename iType>
-  KOKKOS_INLINE_FUNCTION
-      std::enable_if_t<std::is_same_v<typename drvtraits::value_type,
-                                      typename drvtraits::scalar_array_type> &&
-                           std::is_integral_v<iType>,
-                       reference_type>
-      operator[](const iType& i0) const {
+  KOKKOS_INLINE_FUNCTION std::enable_if_t<
+      std::is_integral_v<iType> &&
+          std::is_same_v<typename drvtraits::value_type,
+                         typename drvtraits::scalar_array_type>,
+      reference_type>
+  operator[](const iType& i0) const {
     // Phalanx is violating this, since they use the operator to access ALL
     // elements in the allocation KOKKOS_IMPL_VIEW_OPERATOR_VERIFY( (1 ,
     // this->rank(), m_track, m_map) )
@@ -620,21 +620,19 @@ class DynRankView : public ViewTraits<DataType, Properties...> {
 
   // Rank 1 parenthesis
   template <typename iType>
-  KOKKOS_INLINE_FUNCTION
-      std::enable_if_t<(std::is_void_v<typename traits::specialize> &&
-                        std::is_integral_v<iType>),
-                       reference_type>
-      operator()(const iType& i0) const {
+  KOKKOS_INLINE_FUNCTION std::enable_if_t<
+      std::is_integral_v<iType> && std::is_void_v<typename traits::specialize>,
+      reference_type>
+  operator()(const iType& i0) const {
     KOKKOS_IMPL_VIEW_OPERATOR_VERIFY((1, this->rank(), m_track, m_map, i0))
     return m_map.reference(i0);
   }
 
   template <typename iType>
-  KOKKOS_INLINE_FUNCTION
-      std::enable_if_t<!(std::is_void_v<typename traits::specialize> &&
-                         std::is_integral_v<iType>),
-                       reference_type>
-      operator()(const iType& i0) const {
+  KOKKOS_INLINE_FUNCTION std::enable_if_t<
+      std::is_integral_v<iType> && !std::is_void_v<typename traits::specialize>,
+      reference_type>
+  operator()(const iType& i0) const {
     KOKKOS_IMPL_VIEW_OPERATOR_VERIFY((1, this->rank(), m_track, m_map, i0))
     return m_map.reference(i0, 0, 0, 0, 0, 0, 0);
   }
@@ -642,8 +640,8 @@ class DynRankView : public ViewTraits<DataType, Properties...> {
   // Rank 2
   template <typename iType0, typename iType1>
   KOKKOS_INLINE_FUNCTION std::enable_if_t<
-      (std::is_void_v<typename traits::specialize> &&
-       std::is_integral_v<iType0> && std::is_integral_v<iType1>),
+      std::is_integral_v<iType0> && std::is_integral_v<iType1> &&
+          std::is_void_v<typename traits::specialize>,
       reference_type>
   operator()(const iType0& i0, const iType1& i1) const {
     KOKKOS_IMPL_VIEW_OPERATOR_VERIFY((2, this->rank(), m_track, m_map, i0, i1))
@@ -651,11 +649,11 @@ class DynRankView : public ViewTraits<DataType, Properties...> {
   }
 
   template <typename iType0, typename iType1>
-  KOKKOS_INLINE_FUNCTION
-      std::enable_if_t<!(std::is_void_v<typename drvtraits::specialize> &&
-                         std::is_integral_v<iType0>),
-                       reference_type>
-      operator()(const iType0& i0, const iType1& i1) const {
+  KOKKOS_INLINE_FUNCTION std::enable_if_t<
+      std::is_integral_v<iType0> && std::is_integral_v<iType1> &&
+          !std::is_void_v<typename drvtraits::specialize>,
+      reference_type>
+  operator()(const iType0& i0, const iType1& i1) const {
     KOKKOS_IMPL_VIEW_OPERATOR_VERIFY((2, this->rank(), m_track, m_map, i0, i1))
     return m_map.reference(i0, i1, 0, 0, 0, 0, 0);
   }
@@ -663,9 +661,9 @@ class DynRankView : public ViewTraits<DataType, Properties...> {
   // Rank 3
   template <typename iType0, typename iType1, typename iType2>
   KOKKOS_INLINE_FUNCTION std::enable_if_t<
-      (std::is_void_v<typename traits::specialize> &&
-       std::is_integral_v<iType0> && std::is_integral_v<iType1> &&
-       std::is_integral_v<iType2>),
+      std::is_integral_v<iType0> && std::is_integral_v<iType1> &&
+          std::is_integral_v<iType2> &&
+          std::is_void_v<typename traits::specialize>,
       reference_type>
   operator()(const iType0& i0, const iType1& i1, const iType2& i2) const {
     KOKKOS_IMPL_VIEW_OPERATOR_VERIFY(
@@ -674,11 +672,12 @@ class DynRankView : public ViewTraits<DataType, Properties...> {
   }
 
   template <typename iType0, typename iType1, typename iType2>
-  KOKKOS_INLINE_FUNCTION
-      std::enable_if_t<!(std::is_void_v<typename drvtraits::specialize> &&
-                         std::is_integral_v<iType0>),
-                       reference_type>
-      operator()(const iType0& i0, const iType1& i1, const iType2& i2) const {
+  KOKKOS_INLINE_FUNCTION std::enable_if_t<
+      std::is_integral_v<iType0> && std::is_integral_v<iType1> &&
+          std::is_integral_v<iType2> &&
+          !std::is_void_v<typename drvtraits::specialize>,
+      reference_type>
+  operator()(const iType0& i0, const iType1& i1, const iType2& i2) const {
     KOKKOS_IMPL_VIEW_OPERATOR_VERIFY(
         (3, this->rank(), m_track, m_map, i0, i1, i2))
     return m_map.reference(i0, i1, i2, 0, 0, 0, 0);
@@ -687,9 +686,9 @@ class DynRankView : public ViewTraits<DataType, Properties...> {
   // Rank 4
   template <typename iType0, typename iType1, typename iType2, typename iType3>
   KOKKOS_INLINE_FUNCTION std::enable_if_t<
-      (std::is_void_v<typename traits::specialize> &&
-       std::is_integral_v<iType0> && std::is_integral_v<iType1> &&
-       std::is_integral_v<iType2> && std::is_integral_v<iType3>),
+      std::is_integral_v<iType0> && std::is_integral_v<iType1> &&
+          std::is_integral_v<iType2> && std::is_integral_v<iType3> &&
+          std::is_void_v<typename traits::specialize>,
       reference_type>
   operator()(const iType0& i0, const iType1& i1, const iType2& i2,
              const iType3& i3) const {
@@ -699,12 +698,13 @@ class DynRankView : public ViewTraits<DataType, Properties...> {
   }
 
   template <typename iType0, typename iType1, typename iType2, typename iType3>
-  KOKKOS_INLINE_FUNCTION
-      std::enable_if_t<!(std::is_void_v<typename drvtraits::specialize> &&
-                         std::is_integral_v<iType0>),
-                       reference_type>
-      operator()(const iType0& i0, const iType1& i1, const iType2& i2,
-                 const iType3& i3) const {
+  KOKKOS_INLINE_FUNCTION std::enable_if_t<
+      std::is_integral_v<iType0> && std::is_integral_v<iType1> &&
+          std::is_integral_v<iType2> && std::is_integral_v<iType3> &&
+          !std::is_void_v<typename traits::specialize>,
+      reference_type>
+  operator()(const iType0& i0, const iType1& i1, const iType2& i2,
+             const iType3& i3) const {
     KOKKOS_IMPL_VIEW_OPERATOR_VERIFY(
         (4, this->rank(), m_track, m_map, i0, i1, i2, i3))
     return m_map.reference(i0, i1, i2, i3, 0, 0, 0);
@@ -714,10 +714,10 @@ class DynRankView : public ViewTraits<DataType, Properties...> {
   template <typename iType0, typename iType1, typename iType2, typename iType3,
             typename iType4>
   KOKKOS_INLINE_FUNCTION std::enable_if_t<
-      (std::is_void_v<typename traits::specialize> &&
-       std::is_integral_v<iType0> && std::is_integral_v<iType1> &&
-       std::is_integral_v<iType2> && std::is_integral_v<iType3> &&
-       std::is_integral_v<iType4>),
+      std::is_integral_v<iType0> && std::is_integral_v<iType1> &&
+          std::is_integral_v<iType2> && std::is_integral_v<iType3> &&
+          std::is_integral_v<iType4> &&
+          std::is_void_v<typename traits::specialize>,
       reference_type>
   operator()(const iType0& i0, const iType1& i1, const iType2& i2,
              const iType3& i3, const iType4& i4) const {
@@ -728,12 +728,14 @@ class DynRankView : public ViewTraits<DataType, Properties...> {
 
   template <typename iType0, typename iType1, typename iType2, typename iType3,
             typename iType4>
-  KOKKOS_INLINE_FUNCTION
-      std::enable_if_t<!(std::is_void_v<typename drvtraits::specialize> &&
-                         std::is_integral_v<iType0>),
-                       reference_type>
-      operator()(const iType0& i0, const iType1& i1, const iType2& i2,
-                 const iType3& i3, const iType4& i4) const {
+  KOKKOS_INLINE_FUNCTION std::enable_if_t<
+      std::is_integral_v<iType0> && std::is_integral_v<iType1> &&
+          std::is_integral_v<iType2> && std::is_integral_v<iType3> &&
+          std::is_integral_v<iType4> &&
+          !std::is_void_v<typename traits::specialize>,
+      reference_type>
+  operator()(const iType0& i0, const iType1& i1, const iType2& i2,
+             const iType3& i3, const iType4& i4) const {
     KOKKOS_IMPL_VIEW_OPERATOR_VERIFY(
         (5, this->rank(), m_track, m_map, i0, i1, i2, i3, i4))
     return m_map.reference(i0, i1, i2, i3, i4, 0, 0);
@@ -743,10 +745,10 @@ class DynRankView : public ViewTraits<DataType, Properties...> {
   template <typename iType0, typename iType1, typename iType2, typename iType3,
             typename iType4, typename iType5>
   KOKKOS_INLINE_FUNCTION std::enable_if_t<
-      (std::is_void_v<typename traits::specialize> &&
-       std::is_integral_v<iType0> && std::is_integral_v<iType1> &&
-       std::is_integral_v<iType2> && std::is_integral_v<iType3> &&
-       std::is_integral_v<iType4> && std::is_integral_v<iType5>),
+      std::is_integral_v<iType0> && std::is_integral_v<iType1> &&
+          std::is_integral_v<iType2> && std::is_integral_v<iType3> &&
+          std::is_integral_v<iType4> && std::is_integral_v<iType5> &&
+          std::is_void_v<typename traits::specialize>,
       reference_type>
   operator()(const iType0& i0, const iType1& i1, const iType2& i2,
              const iType3& i3, const iType4& i4, const iType5& i5) const {
@@ -757,12 +759,14 @@ class DynRankView : public ViewTraits<DataType, Properties...> {
 
   template <typename iType0, typename iType1, typename iType2, typename iType3,
             typename iType4, typename iType5>
-  KOKKOS_INLINE_FUNCTION
-      std::enable_if_t<!(std::is_void_v<typename drvtraits::specialize> &&
-                         std::is_integral_v<iType0>),
-                       reference_type>
-      operator()(const iType0& i0, const iType1& i1, const iType2& i2,
-                 const iType3& i3, const iType4& i4, const iType5& i5) const {
+  KOKKOS_INLINE_FUNCTION std::enable_if_t<
+      std::is_integral_v<iType0> && std::is_integral_v<iType1> &&
+          std::is_integral_v<iType2> && std::is_integral_v<iType3> &&
+          std::is_integral_v<iType4> && std::is_integral_v<iType5> &&
+          !std::is_void_v<typename traits::specialize>,
+      reference_type>
+  operator()(const iType0& i0, const iType1& i1, const iType2& i2,
+             const iType3& i3, const iType4& i4, const iType5& i5) const {
     KOKKOS_IMPL_VIEW_OPERATOR_VERIFY(
         (6, this->rank(), m_track, m_map, i0, i1, i2, i3, i4, i5))
     return m_map.reference(i0, i1, i2, i3, i4, i5, 0);
@@ -796,21 +800,19 @@ class DynRankView : public ViewTraits<DataType, Properties...> {
   // Rank 1
   // Rank 1 parenthesis
   template <typename iType>
-  KOKKOS_INLINE_FUNCTION
-      std::enable_if_t<(std::is_void_v<typename traits::specialize> &&
-                        std::is_integral_v<iType>),
-                       reference_type>
-      access(const iType& i0) const {
+  KOKKOS_INLINE_FUNCTION std::enable_if_t<
+      std::is_integral_v<iType> && std::is_void_v<typename traits::specialize>,
+      reference_type>
+  access(const iType& i0) const {
     KOKKOS_IMPL_VIEW_OPERATOR_VERIFY((1, this->rank(), m_track, m_map, i0))
     return m_map.reference(i0);
   }
 
   template <typename iType>
-  KOKKOS_INLINE_FUNCTION
-      std::enable_if_t<!(std::is_void_v<typename traits::specialize> &&
-                         std::is_integral_v<iType>),
-                       reference_type>
-      access(const iType& i0) const {
+  KOKKOS_INLINE_FUNCTION std::enable_if_t<
+      std::is_integral_v<iType> && !std::is_void_v<typename traits::specialize>,
+      reference_type>
+  access(const iType& i0) const {
     KOKKOS_IMPL_VIEW_OPERATOR_VERIFY((1, this->rank(), m_track, m_map, i0))
     return m_map.reference(i0, 0, 0, 0, 0, 0, 0);
   }
@@ -818,8 +820,8 @@ class DynRankView : public ViewTraits<DataType, Properties...> {
   // Rank 2
   template <typename iType0, typename iType1>
   KOKKOS_INLINE_FUNCTION std::enable_if_t<
-      (std::is_void_v<typename traits::specialize> &&
-       std::is_integral_v<iType0> && std::is_integral_v<iType1>),
+      std::is_integral_v<iType0> && std::is_integral_v<iType1> &&
+          std::is_void_v<typename traits::specialize>,
       reference_type>
   access(const iType0& i0, const iType1& i1) const {
     KOKKOS_IMPL_VIEW_OPERATOR_VERIFY((2, this->rank(), m_track, m_map, i0, i1))
@@ -827,11 +829,11 @@ class DynRankView : public ViewTraits<DataType, Properties...> {
   }
 
   template <typename iType0, typename iType1>
-  KOKKOS_INLINE_FUNCTION
-      std::enable_if_t<!(std::is_void_v<typename drvtraits::specialize> &&
-                         std::is_integral_v<iType0>),
-                       reference_type>
-      access(const iType0& i0, const iType1& i1) const {
+  KOKKOS_INLINE_FUNCTION std::enable_if_t<
+      std::is_integral_v<iType0> && std::is_integral_v<iType1> &&
+          !std::is_void_v<typename traits::specialize>,
+      reference_type>
+  access(const iType0& i0, const iType1& i1) const {
     KOKKOS_IMPL_VIEW_OPERATOR_VERIFY((2, this->rank(), m_track, m_map, i0, i1))
     return m_map.reference(i0, i1, 0, 0, 0, 0, 0);
   }
@@ -839,9 +841,9 @@ class DynRankView : public ViewTraits<DataType, Properties...> {
   // Rank 3
   template <typename iType0, typename iType1, typename iType2>
   KOKKOS_INLINE_FUNCTION std::enable_if_t<
-      (std::is_void_v<typename traits::specialize> &&
-       std::is_integral_v<iType0> && std::is_integral_v<iType1> &&
-       std::is_integral_v<iType2>),
+      std::is_integral_v<iType0> && std::is_integral_v<iType1> &&
+          std::is_integral_v<iType2> &&
+          std::is_void_v<typename traits::specialize>,
       reference_type>
   access(const iType0& i0, const iType1& i1, const iType2& i2) const {
     KOKKOS_IMPL_VIEW_OPERATOR_VERIFY(
@@ -850,11 +852,12 @@ class DynRankView : public ViewTraits<DataType, Properties...> {
   }
 
   template <typename iType0, typename iType1, typename iType2>
-  KOKKOS_INLINE_FUNCTION
-      std::enable_if_t<!(std::is_void_v<typename drvtraits::specialize> &&
-                         std::is_integral_v<iType0>),
-                       reference_type>
-      access(const iType0& i0, const iType1& i1, const iType2& i2) const {
+  KOKKOS_INLINE_FUNCTION std::enable_if_t<
+      std::is_integral_v<iType0> && std::is_integral_v<iType1> &&
+          std::is_integral_v<iType2> &&
+          !std::is_void_v<typename traits::specialize>,
+      reference_type>
+  access(const iType0& i0, const iType1& i1, const iType2& i2) const {
     KOKKOS_IMPL_VIEW_OPERATOR_VERIFY(
         (3, this->rank(), m_track, m_map, i0, i1, i2))
     return m_map.reference(i0, i1, i2, 0, 0, 0, 0);
@@ -863,9 +866,9 @@ class DynRankView : public ViewTraits<DataType, Properties...> {
   // Rank 4
   template <typename iType0, typename iType1, typename iType2, typename iType3>
   KOKKOS_INLINE_FUNCTION std::enable_if_t<
-      (std::is_void_v<typename traits::specialize> &&
-       std::is_integral_v<iType0> && std::is_integral_v<iType1> &&
-       std::is_integral_v<iType2> && std::is_integral_v<iType3>),
+      std::is_integral_v<iType0> && std::is_integral_v<iType1> &&
+          std::is_integral_v<iType2> && std::is_integral_v<iType3> &&
+          std::is_void_v<typename traits::specialize>,
       reference_type>
   access(const iType0& i0, const iType1& i1, const iType2& i2,
          const iType3& i3) const {
@@ -875,12 +878,13 @@ class DynRankView : public ViewTraits<DataType, Properties...> {
   }
 
   template <typename iType0, typename iType1, typename iType2, typename iType3>
-  KOKKOS_INLINE_FUNCTION
-      std::enable_if_t<!(std::is_void_v<typename drvtraits::specialize> &&
-                         std::is_integral_v<iType0>),
-                       reference_type>
-      access(const iType0& i0, const iType1& i1, const iType2& i2,
-             const iType3& i3) const {
+  KOKKOS_INLINE_FUNCTION std::enable_if_t<
+      std::is_integral_v<iType0> && std::is_integral_v<iType1> &&
+          std::is_integral_v<iType2> && std::is_integral_v<iType3> &&
+          !std::is_void_v<typename traits::specialize>,
+      reference_type>
+  access(const iType0& i0, const iType1& i1, const iType2& i2,
+         const iType3& i3) const {
     KOKKOS_IMPL_VIEW_OPERATOR_VERIFY(
         (4, this->rank(), m_track, m_map, i0, i1, i2, i3))
     return m_map.reference(i0, i1, i2, i3, 0, 0, 0);
@@ -890,10 +894,10 @@ class DynRankView : public ViewTraits<DataType, Properties...> {
   template <typename iType0, typename iType1, typename iType2, typename iType3,
             typename iType4>
   KOKKOS_INLINE_FUNCTION std::enable_if_t<
-      (std::is_void_v<typename traits::specialize> &&
-       std::is_integral_v<iType0> && std::is_integral_v<iType1> &&
-       std::is_integral_v<iType2> && std::is_integral_v<iType3> &&
-       std::is_integral_v<iType4>),
+      std::is_integral_v<iType0> && std::is_integral_v<iType1> &&
+          std::is_integral_v<iType2> && std::is_integral_v<iType3> &&
+          std::is_integral_v<iType4> &&
+          std::is_void_v<typename traits::specialize>,
       reference_type>
   access(const iType0& i0, const iType1& i1, const iType2& i2, const iType3& i3,
          const iType4& i4) const {
@@ -904,12 +908,14 @@ class DynRankView : public ViewTraits<DataType, Properties...> {
 
   template <typename iType0, typename iType1, typename iType2, typename iType3,
             typename iType4>
-  KOKKOS_INLINE_FUNCTION
-      std::enable_if_t<!(std::is_void_v<typename drvtraits::specialize> &&
-                         std::is_integral_v<iType0>),
-                       reference_type>
-      access(const iType0& i0, const iType1& i1, const iType2& i2,
-             const iType3& i3, const iType4& i4) const {
+  KOKKOS_INLINE_FUNCTION std::enable_if_t<
+      std::is_integral_v<iType0> && std::is_integral_v<iType1> &&
+          std::is_integral_v<iType2> && std::is_integral_v<iType3> &&
+          std::is_integral_v<iType4> &&
+          !std::is_void_v<typename traits::specialize>,
+      reference_type>
+  access(const iType0& i0, const iType1& i1, const iType2& i2, const iType3& i3,
+         const iType4& i4) const {
     KOKKOS_IMPL_VIEW_OPERATOR_VERIFY(
         (5, this->rank(), m_track, m_map, i0, i1, i2, i3, i4))
     return m_map.reference(i0, i1, i2, i3, i4, 0, 0);
@@ -919,10 +925,10 @@ class DynRankView : public ViewTraits<DataType, Properties...> {
   template <typename iType0, typename iType1, typename iType2, typename iType3,
             typename iType4, typename iType5>
   KOKKOS_INLINE_FUNCTION std::enable_if_t<
-      (std::is_void_v<typename traits::specialize> &&
-       std::is_integral_v<iType0> && std::is_integral_v<iType1> &&
-       std::is_integral_v<iType2> && std::is_integral_v<iType3> &&
-       std::is_integral_v<iType4> && std::is_integral_v<iType5>),
+      std::is_integral_v<iType0> && std::is_integral_v<iType1> &&
+          std::is_integral_v<iType2> && std::is_integral_v<iType3> &&
+          std::is_integral_v<iType4> && std::is_integral_v<iType5> &&
+          std::is_void_v<typename traits::specialize>,
       reference_type>
   access(const iType0& i0, const iType1& i1, const iType2& i2, const iType3& i3,
          const iType4& i4, const iType5& i5) const {
@@ -933,12 +939,14 @@ class DynRankView : public ViewTraits<DataType, Properties...> {
 
   template <typename iType0, typename iType1, typename iType2, typename iType3,
             typename iType4, typename iType5>
-  KOKKOS_INLINE_FUNCTION
-      std::enable_if_t<!(std::is_void_v<typename drvtraits::specialize> &&
-                         std::is_integral_v<iType0>),
-                       reference_type>
-      access(const iType0& i0, const iType1& i1, const iType2& i2,
-             const iType3& i3, const iType4& i4, const iType5& i5) const {
+  KOKKOS_INLINE_FUNCTION std::enable_if_t<
+      std::is_integral_v<iType0> && std::is_integral_v<iType1> &&
+          std::is_integral_v<iType2> && std::is_integral_v<iType3> &&
+          std::is_integral_v<iType4> && std::is_integral_v<iType5> &&
+          !std::is_void_v<typename traits::specialize>,
+      reference_type>
+  access(const iType0& i0, const iType1& i1, const iType2& i2, const iType3& i3,
+         const iType4& i4, const iType5& i5) const {
     KOKKOS_IMPL_VIEW_OPERATOR_VERIFY(
         (6, this->rank(), m_track, m_map, i0, i1, i2, i3, i4, i5))
     return m_map.reference(i0, i1, i2, i3, i4, i5, 0);

